articles/node_lifecycle.md

```diff
@@ -27,224 +27,447 @@ Last Modified: {% if page.last_modified %}{{ page.last_modified }}{% else %}{{ p
+An ignored cancellation request becomes inactive and failed the instant a transition is inactive.
+There will be no automated retrying or transitioning of an ignored cancellation request.
+A handled cancel will invalidate any asynchronous handle (i.e., the user transition will be considered inactive).
+
+
+#### **Transition State: `Configuring`**
+In this Transition State the node's, the node is expected to load its configuration and conduct any required setup. 
+The node uses this to set up any resources it must hold throughout its life (irrespective of if it is `Active` or `Inactive`).
+Example resources may include memory that is held continuously or initializing configuration parameters.
+
+*Outbound Transitions*
+- `SUCCESS` → `Inactive`
+- `FAILURE` → `Unconfigured`
+- `ERROR` → `ErrorProcessing`
+
+#### **Transition State: `Activating`**
+This state is expected to do any final preparations to start executing.
 This may include acquiring resources that are only held while the node is actually active, such as access to hardware.
-Ideally, no preparation that requires significant time (such as lengthy hardware initialisation) should be performed in this callback.
+Ideally, no preparation that requires significant time (such as lengthy hardware initialization) should be performed in this callback.
+Ideally, activating should avoid failure if at all possible.
+If there is a possibility of failure, it may be worthwhile to consider moving code to the `Configuring` transition.
 
-#### Valid transitions out if Activating
+*Outbound Transitions*
+- `SUCCESS` → `Active`
+- `FAILURE` → `Inactive`
+- `ERROR` → `ErrorProcessing`
 
-- If the `onActivate` callback succeeds the node will transition to `Active`.
-- If the `onActivate` callback raises or results in any other return code the node will transition to `ErrorProcessing`.
+#### **Transition State: `Deactivating`**
+This state is expected to do any cleanup to stop executing, essentially reversing the `Activating` changes.
 
-### Transition State: Deactivating
+*Outbound Transitions*
+- `SUCCESS` → `Inactive`
+- `FAILURE` → `Active`
+- `ERROR` → `ErrorProcessing`
 
-In this transition state the callback `onDeactivate` will be executed.
-This method is expected to do any cleanup to start executing, and should reverse the `onActivate` changes.
+#### **Transition State: `CleaningUp`**
+This state is expected to clear all state and return the node to a functionally equivalent state as when first created, essentially reversing the `Configuring` changes.
 
-#### Valid transitions out of Deactivating
 
-- If the `onDeactivate` callback succeeds the node will transition to `Inactive`.
-- If the `onDeactivate` callback raises or results in any other return code the node will transition to `ErrorProcessing`.
+*Outbound Transitions*
+- `SUCCESS` → `Unconfigured`
+- `FAILURE` → `Active`
```

Comment:
Shouldn't on FAILURE to CleanUp here transition to _Inactive_ and not Active (per the diagram)?

Reply:
@jfinken I haven't looked closely yet but assume all of your review comments are very likely correct. I may be able able to get to them this weekend but also just made you a collaborator if you wanted to make the changes directly